src/react/createConnectedStore.tsx

```diff
@@ -45,23 +45,30 @@ export function createConnectedStore<State extends object>(
         fx(this.storeDefinition)
       }
 
+      this.subscription = this._createSubscription()
       this.state = {
         storeSnapshot: this.storeDefinition.getCurrentSnapshot()
       }
+    }
 
-      this.subscription = this.storeDefinition.onAll().subscribe(() =>
+    _createSubscription = () =>
+      this.storeDefinition.onAll().subscribe(() =>
         this.setState({
           storeSnapshot: this.storeDefinition.getCurrentSnapshot()
         })
       )
+
+    componentDidMount(): void {
+      if (this.subscription == null) {
+        this.subscription = this._createSubscription()
+      }
     }
 
     componentWillUnmount() {
-      this.subscription.unsubscribe()
-      // Let the state get GC'd.
-      // TODO: Find a more elegant way to do this.
-      ;(this.storeDefinition as any).storeSnapshot = null
-      ;(this as any).storeDefinition = null
+      if (this.subscription != null) {
+        this.subscription.unsubscribe()
+        this.subscription = null
```

Comment:
Can we keep the GC lines? These were pretty important for long-running apps with complex state, IIRC.
```suggestion
        this.subscription = null
        // Let the state get GC'd.
        // TODO: Find a more elegant way to do this.
        ;(this.storeDefinition as any).storeSnapshot = null
        ;(this as any).storeDefinition = null
```

Reply:
`componentDidMount` above requires access to the `storeDefinition` to re-establish the subscription. Not sure there's a way around keeping the reference here?